torch/fx/experimental/symbolic_shapes.py

```diff
@@ -4115,18 +4158,15 @@ def _refine_ranges(self, guard: ShapeGuard) -> None:
                 # Strictly greater relations allow us to refine a bit more, since
                 # x < y implies that the lower bound for x is: y + 1.
                 lower = rhs_vr.lower + int(isinstance(r_expr, sympy.Gt))
-                lower_guard = guard
             if upper > rhs_vr.upper and isinstance(r_expr, (sympy.Eq, sympy.Le, sympy.Lt)):
                 upper = rhs_vr.upper - int(isinstance(r_expr, sympy.Lt))
-                upper_guard = guard
 
             # Do nothing if the new value range is no better than what we already have.
             if vr == ValueRanges(lower, upper):
                 continue
 
             # Updates the range and the guards corresponding to each bound of the symbol.
             self.var_to_range[symbol] = ValueRanges(lower, upper)
-            self.var_to_guards[symbol] = (lower_guard, upper_guard)
```

Comment:
The reason it is not necessary (and you can see all the tests passed!) is because we issue dedicated guards for the value range. So in your example, when you have range `[6, ...]` we produce a guard `s0 >= 6`. If I discard a subsequent guard, including `s0 > 5`, it is because they were made strictly redundant by the range specific guard.

When I originally reviewed your PR, what I didn't quite understand was how to deal with things like `s0 < s1` which showed up in `var_to_guards`. Could these be necessary to preserve? But in fact, I have convinced myself that it is not necessary. If this guard is not statically implied by the refined ranges, I won't drop it (as you are worried). If it is statically implied, it is not needed, the other guards + var ranges are enough.

Reply:
That makes a lot of sense. Thank you for explaining it to me thoroughly.